metastore/src/java/org/apache/hadoop/hive/metastore/hbase/ObjectCache.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.hadoop.hive.metastore.hbase;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A generic class for caching objects obtained from HBase.  Currently a set of
+ * convenience methods around a {@link java.util.HashMap} with a max size but built
+ * as a separate class in case we want to switch out the implementation to something more efficient.
```

Comment:
Can you also add comment saying that this cache is not synchrnonized because it is going to be used from only one thread (it is indirectly threadlocal)